types/webpack-dev-server/index.d.ts

```diff
@@ -108,20 +182,11 @@ declare namespace WebpackDevServer {
         /**
          * Specify a host to use. By default this is localhost.
          */
-        host?: string | undefined;
+        host: string;
         /**
          * Enable webpack's Hot Module Replacement feature.
-         * Note that webpack.HotModuleReplacementPlugin is required to fully
-         * enable HMR. If webpack or webpack-dev-server are launched with the
-         * --hot option, this plugin will be added automatically, so you may
-         * not need to add this to your webpack.config.js.
          */
-        hot?: boolean | undefined;
-        /**
-         * Enables Hot Module Replacement (see devServer.hot) without page
-         * refresh as fallback in case of build failures.
-         */
-        hotOnly?: boolean | undefined;
+        hot?: boolean | string | undefined;
```

Comment:
```suggestion
        hot?: boolean | 'only'| undefined;
```

Reply:
I did so originally, but in my project on which I tested the change (`hot: cliOptions.hotOnly ? 'only' : true,`), I got this error:

```
[server]   Types of property 'hot' are incompatible.
[server]     Type 'string | boolean' is not assignable to type 'boolean | "only" | undefined'.
```

That's why I gave up using allowed strings in all types.

Reply:
can you adapt playground, so I can see the problem?:
https://www.typescriptlang.org/play?ts=4.4.0-beta#code/JYOwLgpgTgZghgYwgAgMIHsQ2AcwK5RxjCbIDeAsAFDLIAW6YA-AFzIBG66ANhHCMgA+yAOSZuATxHC8IACYRsICHIDc1AL7VqoSLEQoA8gAdimAM7lqtBmEMhJbc2CigcQjl1791VLVWoYWQQzAQRMbHwoCAAKcKxcVjQI3AIiEhAASjINX2p452R0UwzzNhNQywBeKxp6RntHUXEpTXyUqNiczN94yIIu2zZiyoA6W0aJWiZmhylkNhc8FA1MoA
thx!

Reply:
Updated playground (see https://www.typescriptlang.org/play?ts=4.1.5#code/JYOwLgpgTgZghgYwgAgMIHsQ2AcwK5RxjCbIDeAsAFDLIAW6YA-AFzIBG66ANhHCMgA+yAOSZuATxHC8IACYRsICHIDc1AL7VqoSLEQoA8gAdimAM7lqtBmEMhJbTjz4CZ8xaBXqqWqtRhZBDMBBExsfCgIAAowrFxWNHDcAiISEABKMg0fajjzMGR0U3TzNhMQywBeKxp6RntHZDAoPAgAGk085MiY7IyfOIiCPts2YsqAOltGiVomUXEpZDYWtuQNDKA) -- although there are error, but I have it locally (on TS 4.1.5):

[server] Types of property 'hot' are incompatible.
[server] Type 'string | boolean' is not assignable to type 'boolean | "only" | undefined'.
[server] Type 'string' is not assignable to type 'boolean | 'only' | undefined'.

:man_shrugging:

Reply:
As it turned out, my mistake was not having `WebpackDevServer.Configuration` type on config object. So I commit your suggested change, thanks.